internal/provider/common/role.go

```diff
@@ -0,0 +1,70 @@
+package common
```

Comment:
added this file for shared role functions

needed to create a new directory because if i had it in utils i would hit an `import cycle` error since models, resources and utils reference each other in a cycle